app/assets/javascripts/discourse/app/components/user-field.js

```diff
@@ -1,11 +1,33 @@
 import discourseComputed from "discourse-common/utils/decorators";
 import Component from "@ember/component";
 import { fmt } from "discourse/lib/computed";
+import { action } from "@ember/object";
 
 export default Component.extend({
   classNameBindings: [":user-field", "field.field_type", "customFieldClass"],
   layoutName: fmt("field.field_type", "components/user-fields/%@"),
 
+  didInsertElement() {
+    this.field.component = this;
+  },
+
+  @action
+  focus() {
```

Comment:
We would need a more descriptive name for the action please, it's doesn't clash with native ember events (focusIn, focusOut) but still it would be safer.

Reply:
I'm using the name `focus` since I have to call it [here](https://github.com/discourse/discourse/pull/9643/commits/962ef37108252288700981a3c7a89aba446a36b0#diff-092d88170d619c3d0d097005cfa089bbR319). I need to add one more if condition if I change it.